internal/resources/app_endpoint_oidc.go

```diff
@@ -0,0 +1,387 @@
+		diags = resp.State.Set(ctx, plan)
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	r.mapResponseToState(&plan, details)
+
+	diags = resp.State.Set(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+}
+
+// Delete deletes the OIDC provider.
+func (r *AppEndpointOidcProvider) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state providerschema.AppEndpointOidcProvider
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	organizationId := state.OrganizationId.ValueString()
+	projectId := state.ProjectId.ValueString()
+	clusterId := state.ClusterId.ValueString()
+	appServiceId := state.AppServiceId.ValueString()
+	appEndpointName := state.AppEndpointName.ValueString()
+	providerId := state.ProviderId.ValueString()
+
+	if providerId == "" {
+		return
+	}
+
+	url := fmt.Sprintf(
+		"%s/v4/organizations/%s/projects/%s/clusters/%s/appservices/%s/appEndpoints/%s/oidcProviders/%s",
+		r.HostURL,
+		organizationId,
+		projectId,
+		clusterId,
+		appServiceId,
+		appEndpointName,
+		providerId,
+	)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodDelete, SuccessStatus: http.StatusAccepted}
+	_, err := r.Client.ExecuteWithRetry(ctx, cfg, nil, r.Token, nil)
+	if err != nil {
+		resourceNotFound, _ := api.CheckResourceNotFoundError(err)
+		if resourceNotFound {
+			tflog.Info(ctx, "resource doesn't exist in remote server removing resource from state file")
+			return
+		}
+		resp.Diagnostics.AddError("Error Deleting OIDC Provider", api.ParseError(err))
+		return
+	}
+}
+
+// getOidcProvider gets OIDC provider details.
+func (r *AppEndpointOidcProvider) getOidcProvider(ctx context.Context, organizationId, projectId, clusterId, appServiceId, appEndpointName, providerId string) (api.AppEndpointOIDCProviderResponse, error) {
+	url := fmt.Sprintf(
+		"%s/v4/organizations/%s/projects/%s/clusters/%s/appservices/%s/appEndpoints/%s/oidcProviders/%s",
+		r.HostURL,
+		organizationId,
+		projectId,
+		clusterId,
+		appServiceId,
+		appEndpointName,
+		providerId,
+	)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodGet, SuccessStatus: http.StatusOK}
+	res, err := r.Client.ExecuteWithRetry(ctx, cfg, nil, r.Token, nil)
+	if err != nil {
+		return api.AppEndpointOIDCProviderResponse{}, fmt.Errorf("%s: %w", errors.ErrExecutingRequest, err)
+	}
+	var out api.AppEndpointOIDCProviderResponse
+	if err := json.Unmarshal(res.Body, &out); err != nil {
+		return api.AppEndpointOIDCProviderResponse{}, fmt.Errorf("failed to parse response: %w", err)
+	}
+	return out, nil
+}
+
+// mapResponseToState maps response fields to state.
+// If preserveNulls is true, optional attributes that are null in state will not be populated.
+func (r *AppEndpointOidcProvider) mapResponseToState(state *providerschema.AppEndpointOidcProvider, resp api.AppEndpointOIDCProviderResponse) {
+	// Required fields
+	if resp.Issuer != "" {
```

Comment:
all these if checks aren't needed.  just set the state value

Reply:
https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/396/commits/99c3b3f671e53c7ef50fb0fba9973f3950704793